src/core/response/parameters.rs

```diff
@@ -46,28 +52,96 @@ impl From<VpToken> for Json {
     }
 }
 
+impl VpToken {
+    /// Parse the VP Token as a JSON object.
+    ///
+    /// This will attempt to decode the token as base64, and if that fails, it
+    /// will attempt to parse the token as a JSON object.
+    ///
+    /// See: https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#section-6.1-2.2
+    ///
+    /// If you want to check for decode errors, use [VpToken::decode_base64].
+    pub fn parse(&self) -> Result<Json, Error> {
+        match self.decode_base64() {
+            Ok(decoded) => Ok(decoded),
+            Err(_) => Ok(serde_json::from_str(&self.0)?),
+        }
+    }
+
+    /// Return the Verifiable Presentation Token as a JSON object from a base64
+    /// encoded string.
+    pub fn decode_base64(&self) -> Result<Json, Error> {
+        let decoded = BASE64_STANDARD.decode(&self.0)?;
+        Ok(serde_json::from_slice(&decoded)?)
+    }
+
+    /// Validate an unencoded Verifiable Presentation Token.
+    ///
+    /// This method assumtes the VP token is not encoded as a JWT.
+    ///
+    /// # Returns
+    ///
+    /// This method will return `Ok(())` if the VP token is valid.
+    ///
+    /// # Errors
+    ///
+    /// This method will return an error if the VP token is invalid,
+    /// or if the verifiable presentation is invalid or does not meet
+    /// the requirements of the presentation definition.
+    ///
+    ///
+    pub fn validate_unencoded(
```

Comment:
As above, this is making an assumption that the VpToken is a VCDM VP. There's also some confusion here about encoded as a JWT vs unencoded. Encoded as a JWT refers to the authorization response (see [direct_post.jwt](https://openid.net/specs/openid-4-verifiable-presentations-1_0-20.html#name-response-mode-direct_postjw)), not the VP token. The VP token is never encoded directly as a JWT. It's either a String, a JSON object, or an array of VP tokens.

Reply:
This method has been removed.